pkg/security/san_pinning.go

```diff
@@ -0,0 +1,183 @@
+package security
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/go-logr/logr"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1alpha1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/runtime-watcher/listener/pkg/types"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	debugLogLevel = 2
```

Comment:
Hey there, well this turned out to be a weird one. We unified the logging structure back in another PR that removed these flags by accident:

The main.go should actually have something like this:

```go
func loggerOptions() *ctrlzap.Options {

	// The following settings is based on kyma community Improvement of log messages usability
	//nolint:lll
	// https://github.com/kyma-project/community/blob/main/concepts/observability-consistent-logging/improvement-of-log-messages-usability.md#log-structure
	opts := &ctrlzap.Options{
		ZapOpts: []zap.Option{
			zap.AddCaller(),
			zap.Fields(zap.Namespace("context")),
		},
		EncoderConfigOptions: []ctrlzap.EncoderConfigOption{
			func(config *zapcore.EncoderConfig) {
				config.TimeKey = "date"
				config.EncodeLevel = zapcore.CapitalLevelEncoder
			},
		},
	}
	opts.BindFlags(flag.CommandLine)

	return opts
}
```
notice the `opts.BindFlags` and not hardcoded level or time config. The reason for this is that before our adaptation of https://github.com/kyma-project/community/blob/main/concepts/observability-consistent-logging/improvement-of-log-messages-usability.md#log-structure they were configured dynamically through the mentioned flags of my previous comment but in the current main that's no longer possible. I would suggest @ruanxin to look into this and make them dynamic again but adjust the default to the consistent structure since he adjusted the implementation. However since we later on want to make debug level configurable I think we anyhow need some more invest here.

For you though that means you can safely change it anyhow.

Reply:
But I used here the `Logr.Logger` since I wanted to use the same as we do in other parts of the lifecycle manager to stay consistent, and now we also have the Zap Logger.

I think in general, we need to reiterate over our loggers to have a consistent logging.